e2e/pages/adf/content-services/documentListPage.ts

```diff
@@ -0,0 +1,117 @@
+    versionManagerAction = element(by.css('button[data-automation-id*="VERSIONS"]'));
+    moveContentElement = element(by.css('button[data-automation-id*="MOVE"]'));
+    copyContentElement = element(by.css('button[data-automation-id*="COPY"]'));
+    lockContentElement = element(by.css('button[data-automation-id="DOCUMENT_LIST.ACTIONS.LOCK"]'));
+    downloadContent = element(by.css('button[data-automation-id*="DOWNLOAD"]'));
+    actionMenu = element(by.css('div[role="menu"]'));
+    optionButton = by.css('button[data-automation-id*="action_menu_"]');
+    rows = by.css('div[id="document-list-container"] div[class*="adf-datatable-body"] div[class*="adf-datatable-row"]');
+
+    dataTablePage() {
+        return new DataTablePage();
+    }
+
+    deleteContent(content) {
+        this.clickOnActionMenu(content);
+        this.waitForContentOptions();
+        this.deleteContentElement.click();
+    }
+
+    checkDeleteIsDisabled(content) {
+        this.clickOnActionMenu(content);
+        this.waitForContentOptions();
+        let disabledDelete = element(by.css(`button[data-automation-id*='DELETE'][disabled='true']`));
+        Util.waitUntilElementIsVisible(disabledDelete);
+    }
+
+    metadataContent(content) {
+        this.clickOnActionMenu(content);
+        this.waitForContentOptions();
+        this.metadataAction.click();
+    }
+
+    versionManagerContent(content) {
+        this.clickOnActionMenu(content);
+        this.waitForContentOptions();
+        this.versionManagerAction.click();
+    }
+
+    copyContent(content) {
+        this.clickOnActionMenu(content);
+        this.copyContentElement.click();
+    }
+
+    lockContent(content) {
+        this.clickOnActionMenu(content);
+        this.lockContentElement.click();
+    }
+
+    waitForContentOptions() {
+        Util.waitUntilElementIsVisible(this.copyContentElement);
+        Util.waitUntilElementIsVisible(this.moveContentElement);
+        Util.waitUntilElementIsVisible(this.deleteContentElement);
+        Util.waitUntilElementIsVisible(this.downloadContent);
+    }
+
+    clickOnActionMenu(content) {
+        this.dataTable.getRowByRowName(content).element(this.optionButton).click();
+        Util.waitUntilElementIsVisible(this.actionMenu);
+        browser.sleep(500);
+        return this;
+    }
+
+    navigateToFolder(folder) {
+        this.dataTable.doubleClickRow(folder);
+        return this;
+    }
+
+    checkContextActionIsVisible(actionName) {
+        let actionButton = element(by.css(`button[data-automation-id="context-${actionName}"`));
+        Util.waitUntilElementIsVisible(actionButton);
+        Util.waitUntilElementIsClickable(actionButton);
+        return actionButton;
+    }
+
+    pressContextMenuActionNamed(actionName) {
+        let actionButton = this.checkContextActionIsVisible(actionName);
+        actionButton.click();
+    }
+
+    checkLockedIcon(content) {
```

Comment:
checkLockedIcon and checkUnlockedIcon should be part of datatable also, but this information could be checked with a developer from the team.